plugin-server/tests/worker/ingestion/person-state.test.ts

```diff
@@ -65,606 +68,1719 @@ describe('PersonState.update()', () => {
+                    version: 1,
+                    is_identified: false,
+                })
+            )
+
+            // verify Postgres distinct_ids
+            const distinctIds = await hub.db.fetchDistinctIdValues(persons[0])
+            expect(distinctIds).toEqual(expect.arrayContaining(['new-user']))
 
-        const clickhousePersons = await delayUntilEventIngested(fetchPersonsRows)
-        expect(clickhousePersons.length).toEqual(2)
-        expect(clickhousePersons).toEqual(
-            expect.arrayContaining([
+            // verify ClickHouse persons
+            const clickhousePersons = await delayUntilEventIngested(fetchPersonsRows)
+            expect(clickhousePersons.length).toEqual(1)
+            expect(clickhousePersons[0]).toEqual(
                 expect.objectContaining({
                     id: uuid.toString(),
-                    properties: JSON.stringify({ a: 1, b: 2 }),
-                    is_deleted: 1,
-                    version: 100,
-                }),
+                    properties: JSON.stringify({ b: 4, c: 4, e: 4 }),
+                    created_at: timestampch,
+                    version: 1,
+                    is_identified: 0,
+                })
+            )
+
+            // verify ClickHouse distinct_ids
+            await delayUntilEventIngested(fetchDistinctIdsRows)
+            const clickHouseDistinctIds = await hub.db.fetchDistinctIdValues(persons[0], Database.ClickHouse)
+            expect(clickHouseDistinctIds).toEqual(expect.arrayContaining(['new-user']))
+
+            // verify personContainer
+            expect(persons[0]).toEqual(await personContainer.get())
+        })
+
+        it('does not update person if no change to is_identified nor properties', async () => {
+            await hub.db.createPerson(timestamp, { b: 3, c: 4 }, {}, {}, 2, null, true, uuid.toString(), [
+                'new-user',
+                'old-user',
+            ])
+
+            const personContainer = await personState({
+                event: '$identify',
+                distinct_id: 'new-user',
+                properties: {
+                    $anon_distinct_id: 'old-user',
+                    $set_once: { c: 3 },
+                    $set: { b: 3 },
+                },
+            }).update()
+            await hub.db.kafkaProducer.flush()
+
+            // verify Postgres persons
+            const persons = await hub.db.fetchPersons()
+            expect(persons.length).toEqual(1)
+            expect(persons[0]).toEqual(
                 expect.objectContaining({
-                    id: mergedPerson!.uuid,
-                    properties: JSON.stringify({ a: 1, b: 4, c: 3, e: 4 }),
-                    is_deleted: 0,
+                    id: expect.any(Number),
+                    uuid: uuid.toString(),
+                    properties: { b: 3, c: 4 },
+                    created_at: timestamp,
+                    version: 0, // note version doesn't change
```

Comment:
Note: We have so much comments per test that this comment just gets lost in the noise. Is this an important aspect of the test?

Reply:
consider checking just the version in this case